op-deployer/pkg/deployer/integration_test/cli/shared.go

```diff
@@ -0,0 +1,83 @@
+package cli
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/devkeys"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/pipeline"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/state"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+// addrFor generates an address for a given role
+func addrFor(t *testing.T, dk *devkeys.MnemonicDevKeys, key devkeys.Key) common.Address {
+	addr, err := dk.Address(key)
+	require.NoError(t, err)
+	return addr
+}
+
+// newChainIntent creates a new chain intent with the given parameters
+func newChainIntent(t *testing.T, dk *devkeys.MnemonicDevKeys, l1ChainID *big.Int, chainID *uint256.Int) state.ChainIntent {
+	return state.ChainIntent{
+		ID:                         chainID.Bytes32(),
+		BaseFeeVaultRecipient:      addrFor(t, dk, devkeys.BaseFeeVaultRecipientRole.Key(l1ChainID)),
+		L1FeeVaultRecipient:        addrFor(t, dk, devkeys.L1FeeVaultRecipientRole.Key(l1ChainID)),
+		SequencerFeeVaultRecipient: addrFor(t, dk, devkeys.SequencerFeeVaultRecipientRole.Key(l1ChainID)),
+		GasLimit:                   60_000_000,
+		Roles: state.ChainRoles{
+			L1ProxyAdminOwner: addrFor(t, dk, devkeys.L2ProxyAdminOwnerRole.Key(l1ChainID)),
+			L2ProxyAdminOwner: addrFor(t, dk, devkeys.L2ProxyAdminOwnerRole.Key(l1ChainID)),
+			SystemConfigOwner: addrFor(t, dk, devkeys.SystemConfigOwner.Key(l1ChainID)),
+			UnsafeBlockSigner: addrFor(t, dk, devkeys.SequencerP2PRole.Key(l1ChainID)),
+			Batcher:           addrFor(t, dk, devkeys.BatcherRole.Key(l1ChainID)),
+			Proposer:          addrFor(t, dk, devkeys.ProposerRole.Key(l1ChainID)),
+			Challenger:        addrFor(t, dk, devkeys.ChallengerRole.Key(l1ChainID)),
+		},
+	}
+}
+
+// defaultPrivkey returns the default private key for testing
+func defaultPrivkey(t *testing.T) (string, *ecdsa.PrivateKey, *devkeys.MnemonicDevKeys) {
```

Comment:
These same functions are already in [apply_test.go](https://github.com/ethereum-optimism/optimism/blob/feat/op-deployer/cli-based-integration-tests/op-deployer/pkg/deployer/integration_test/apply_test.go#L61-L70). Could we de-dupe?